homeassistant/components/rachio/switch.py

```diff
@@ -179,6 +185,72 @@ async def async_added_to_hass(self):
         )
 
 
+class RachioRainDelay(RachioSwitch):
+    """Representation of a rain delay status/switch."""
+
+    def __init__(self, controller):
+        """Instantiate a new Rachio rain delay switch."""
+        super().__init__(controller, poll=True)
+        self._poll_update(controller.init_data)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the switch."""
+        return f"{self._controller.name} rain delay"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique id by combining controller id and purpose."""
+        return f"{self._controller.controller_id}-delay"
+
+    @property
+    def icon(self) -> str:
+        """Return an icon for rain delay."""
+        return "mdi:camera-timer"
+
+    def _poll_update(self, data=None) -> bool:
+        """Request the state from the API."""
+        # API returns either 0 or current UNIX time when rain delay was canceled
+        # depending if it was done from the app or via the API
+        if data is None:
+            data = self._controller.rachio.device.get(self._controller.controller_id)[1]
```

Comment:
Is this async safe?

Reply:
Nope, its also in `RachioStandbySwitch` as well

Reply:
@brg468 Let me know if help fixing this.

Reply:
> @brg468 Let me know if help fixing this.

Honesty I don't know where to begin on this one. As you can tell I based this switch on `RachioStandbySwitch`.

Reply:
Looks like `RachioZone` and `RachioSchedule` all have the same issue so it has been an issue since before I started working on it.

Reply:
Actually those are ok since they access the cache.

Reply:
Did some testing and it never actually does any I/O because it always has data.  I'm going to clean up the side effects in init though as much of this code can go away!

Reply:
https://github.com/home-assistant/core/pull/34799